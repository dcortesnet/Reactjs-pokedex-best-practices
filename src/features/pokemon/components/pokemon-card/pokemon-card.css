.card {
  background-color: #f0f0f0;
  width: 100%;
  height: 100%;
  flex-basis: 100%;
  margin-inline-end: 20px;
  margin-inline-start: 20px;
  margin-block-start: 5px;
  margin-block-end: 5px;
}

.card__flex {
  width: 100%;
  display: flex;
  justify-content: center;
}

@media screen and (min-width: 460px) {
  .card {
    flex-basis: 40%;
    margin-inline-end: 0;
    margin-inline-start: 0;
  }
}

@media screen and (min-width: 765px) {
  .card {
    flex-basis: 15%;
  }
}

.card__body {
  margin-top: 100px;
  height: 50px;
}

.card__pokemon__img {
  object-fit: fill;
  width: 160px;
  height: 160px;
}

.card__background {
  display: flex;
  background: #f00000;
  margin: auto;
  width: 100%;
  height: 120px;
  max-height: 120px;
  border-radius: 5px 5px 0 0;
}

.card__line__img {
  display: flex;
  margin: auto;
  width: 100%;
  height: 5px;
  background: #222224;
}

.card__name {
  text-align: center;
  margin: auto;
  font-size: 1.5em;
}

.card__image {
  margin-top: 40px;
  position: absolute;
  background-color: #f0f0f0;
  width: 160px;
  border-radius: 50%;
  border: 5px solid #222224;
  z-index: 1;
}

.card__type__title {
  text-align: center;
  font-size: 20px;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.card__type__container {
  text-align: center;
  display: inline-flex;
  width: 100%;
}

.card__types {
  border-radius: 10px;
  margin: 15px;
  background: darkcyan;
  width: 100%;
}
