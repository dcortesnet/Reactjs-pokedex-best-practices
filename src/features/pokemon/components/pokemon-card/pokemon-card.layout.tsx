import React from "react";
import { Pokemon } from "../../interfaces/pokemon.interface";
import "./pokemon-card.css";

export const PokemonCardLayout: React.FC<Pokemon> = ({ sprites, name }) => {
  return (
    <div className="card">
      <div className="card__flex">
        <div className="card__image">
          <img
            src={
              sprites.other.dream_world.front_default || sprites.front_default
            }
            alt={name}
            className="card__pokemon__img"
          />
        </div>
      </div>
      <div className="card__background" />
      <div className="card__line__img" />
      <div className="card__body">
        <p className="card__name">{name}</p>
      </div>
    </div>
  );
};
