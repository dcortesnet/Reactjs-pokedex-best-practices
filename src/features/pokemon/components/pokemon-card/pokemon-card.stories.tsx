import React from "react";
import { Story, Meta } from "@storybook/react";
import { pokemonMock } from "../../mocks/pokemon.mock";
import { PokemonCardLayout } from "./pokemon-card.layout";

export default {
  title: "Pokemon/components/PokemonCard",
  component: PokemonCardLayout,
} as Meta;

const Template: Story<any> = (args) => <PokemonCardLayout {...args} />;
export const template = Template.bind(pokemonMock);
template.args = pokemonMock;
