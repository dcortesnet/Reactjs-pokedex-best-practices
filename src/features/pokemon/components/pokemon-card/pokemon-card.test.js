import { PokemonCardLayout } from "./pokemon-card.layout";
import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { pokemonMock } from "../../mocks/pokemon.mock";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });
describe("PokemonCardLayout", () => {
  it("Renders correctly", () => {
    const r = shallow(
      <PokemonCardLayout
        sprites={pokemonMock.sprites}
        types={pokemonMock.types}
      ></PokemonCardLayout>
    );

    expect(toJson(r)).toMatchSnapshot();
  })
});
