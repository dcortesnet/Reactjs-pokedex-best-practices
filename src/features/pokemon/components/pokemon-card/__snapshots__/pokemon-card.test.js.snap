// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCardLayout Renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card__flex"
  >
    <div
      className="card__image"
    >
      <img
        className="card__pokemon__img"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
      />
    </div>
  </div>
  <div
    className="card__background"
  />
  <div
    className="card__line__img"
  />
  <div
    className="card__body"
  >
    <p
      className="card__name"
    />
  </div>
</div>
`;
